datadog_checks_base/datadog_checks/base/stubs/aggregator.py

```diff
@@ -293,6 +293,46 @@ def assert_all_metrics_covered(self):
             missing_metrics = self.not_asserted()
         assert self.metrics_asserted_pct >= 100.0, 'Missing metrics: {}'.format(missing_metrics)
 
+    def assert_no_duplicate_all(self):
+        self.assert_no_duplicate_metrics()
+        self.assert_no_duplicate_service_checks()
+
+    def assert_no_duplicate_metrics(self):
+        metric_stubs = [m for metrics in self._metrics.values() for m in metrics]
+
+        def stub_to_key_fn(stub):
+            return stub.name, stub.type, str(sorted(stub.tags)), stub.hostname
```

Comment:
`count`s are designed to be submitted multiple times

Reply:
Seems in most cases `count`s are only submitted once too.

I can added an exclusion parameter to `assert_no_duplicate_metrics(self, exclude_metrics=None)`.
`exclude_metrics` a list of metrics to not check for duplication.

What do you think ?

Reply:
That option is a good idea, but we need to handle counts please 😄 

Literally the entire point of `count` is that you can submit multiple values during one check run and the Agent will aggregate them to the proper context, see https://github.com/DataDog/integrations-core/blob/ca299bd847e70c82ac3ab9bd0ccef82a528f9c0a/datadog_checks_base/datadog_checks/base/stubs/aggregator.py#L236-L239

Reply:
> we need to handle counts please 

What do you mean ?

I think the `exclude_metrics` is enough to handle the case since submitting count multiple times only happen in few cases, and we can just exclude those metrics form the duplication search.

Reply:
But imo opinion it should happen by default since it's known behavior that will never change. It'll make it easier for us and contributors

Reply:
Updated. Metric types that intended to be called multiple times are now ignored: `[self.COUNT, self.MONOTONIC_COUNT, self.COUNTER]`